sphinx/writers/texinfo.py

```diff
@@ -545,7 +545,10 @@ def add_anchor(self, id: str, node: Node) -> None:
     def add_xref(self, id: str, name: str, node: Node) -> None:
         name = self.escape_menu(name)
         sid = self.get_short_id(id)
-        self.body.append('@ref{%s,,%s}' % (sid, name))
+        if self.config.texinfo_emit_document_references:
+            self.body.append('@ref{%s,,%s}' % (sid, name))
+        else:
+            self.body.append(name)
         self.referenced_ids.add(sid)
         self.referenced_ids.add(self.escape_id(id))
```

Comment:
`@anchor` tags are still needed for these xrefs even if `texinfo_emit_document_references` is True?

Reply:
No, they are not needed, fixed.